annif/project.py

```diff
@@ -153,6 +155,16 @@ def vocab(self):
     def subjects(self):
         return self.vocab.subjects
 
+    @property
+    def is_trained(self):
```

Comment:
I see a couple of issues in implementing is_trained on the project level:
1. How to deal with ensembles (as you mentioned) - in my view, an ensemble could be considered trained when all its sources have been trained, but that requires peeking into the internals of the ensemble
2. External services such as Maui and another instance of Annif used via the `http` backend - they (probably) don't have any local state

I'd rather implement is_trained in the individual backends. The default implementation could be like this, but ensembles and remote service backends could define their own implementation.